### PR TITLE
Mimir 698 dataset repo

### PR DESCRIPTION
**Pull request by Garlov**
MIMIR-698

### CODE REVIEW

File: docs/RepoAndNode.md [hunk 1]
```diff
@@ -17,3 +17,122 @@ After a repo is created, you can store data in it. The data is stored as `nodes`
...
  
+## Dataset
+automatically sets up `no.ssb.dataset` repo and one child node for each of the different dataset sources.
+
+### `getDataset()`
+Example
+```typescript
+const myDataset: DatasetRepoNode<MyDatasetType> = getDataset(DataSource.TALLBYGGER, '123456')
+```
+Returns (undefined if the node with the key doesnt exist)
```
Comment (prabu-ssb) [inline docs/RepoAndNode.md:28]: Could we clip the result from `get` to show only the relevant fields (leave out XP node metadata)  for readability ?

File: src/main/resources/lib/repo/common.ts [hunk 2]
```diff
@@ -101,6 +101,19 @@ export function getChildNodes(repository: string, branch: string, key: string):
 
 export function nodeExists(repository: string, branch: string, key: string): boolean {
   return withConnection(repository, branch, (conn) => {
-    return conn.exists(key).length === 1
+    return !!conn.exists(key)
   })
 }
+
+export interface RepoCommonLib {
```
Comment (prabu-ssb) [inline src/main/resources/lib/repo/common.ts:108]: neat! :-)

File: docs/RepoAndNode.md (hunk 1 above)
Comment (Garlov) [reply docs/RepoAndNode.md:28]: The thought was to do that in the ssb/dataset lib, or somewhere else. So when you're using the repo/dataset lib, you're getting the full node

Comment (prabu-ssb) [reply docs/RepoAndNode.md:28]: Makes sense. I will go through the ssb::statreg doc and see if I can make the changes in one of my subsequent branches/PRs

**Pull request merged**